### PR TITLE
Datasets educational modal restyling

### PR DESCRIPTION
**Pull request by mazameli**
@kdoh did some design explorations to make this a bit nicer, and this PR implements that. Only thing I couldn't get to work was the stupid flex justify-between to make the cancel button go to the left, Kyle. :b

# Before
image

# After
image

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/NewDatasetModal/NewDatasetModal.jsx [hunk 1]
```diff
@@ -30,33 +33,37 @@ function NewDatasetModal({ turnQuestionIntoDataset, onClose }) {
...
+            great starting points for exploration.
+          </DatasetValueProp>
+          <DatasetValueProp>
+            Show up higher in search results and get highlighted when other
+            users start new questions to promote reuse.
+          </DatasetValueProp>
+          <DatasetValueProp>
+            Live in collections to keep them separate from messy database
+            schemas.
+          </DatasetValueProp>
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/components/NewDatasetModal/NewDatasetModal.jsx:64]: We need to wrap the copy with `t`, so they can be translated

**Review by kulyk — CHANGES_REQUESTED** [review]
Really like the new look and the fact that the modal now tells you more about why you should create datasets 👍  The only thing I noticed are missing translations

File: frontend/src/metabase/query_builder/components/NewDatasetModal/NewDatasetModal.jsx (hunk 1 above)
Comment (mazameli) [reply frontend/src/metabase/query_builder/components/NewDatasetModal/NewDatasetModal.jsx:64]: Whoops! Totally forgot, thanks!

**Review by kulyk — APPROVED** [review]
🚀

**Pull request merged**